File: go/PointList/app/pointlist/point.go
package pointlist

import "fmt"

type Point struct {
	xpos int
	ypos int
}

func (ppos *Point) SetPointPos(xpos int, ypos int) {
	ppos.xpos = xpos
	ppos.ypos = ypos
}

func (ppos *Point) ShowPointPos() {
	fmt.Printf("[%d, %d] \n", ppos.xpos, ppos.ypos)
}

func PointComp(pos1 *Point, pos2 *Point) int {
	if pos1.xpos == pos2.xpos && pos1.ypos == pos2.ypos {
		return 0
	} else if pos1.xpos == pos2.xpos {
		return 1
	} else if pos1.ypos == pos2.ypos {
		return 2
	} else {
		return -1
	}
}

File: python/chapter3/PointList/app/pointlist/point.py
from pydantic import BaseModel

class Point(BaseModel):
    xpos: int = 0
    ypos: int = 0

    def SetPointPos(self, xpos: int, ypos: int):
        self.xpos = xpos
        self.ypos = ypos

    def ShowPointPos(self):
        print(f"[{self.xpos}, {self.ypos}]")    

def PointComp(pos1: Point, pos2: Point):
    if pos1.xpos == pos2.xpos and pos1.ypos == pos2.ypos:
        return 0
    elif pos1.xpos == pos2.xpos:
        return 1
    elif pos1.ypos == pos2.ypos:
        return 2
    else:
        return -1